geode-protobuf/src/test/java/org/apache/geode/internal/protocol/protobuf/v1/acceptance/LocatorConnectionAuthenticationDUnitTest.java

```diff
@@ -0,0 +1,167 @@
+ */
+@Category(DistributedTest.class)
+public class LocatorConnectionAuthenticationDUnitTest extends JUnit4CacheTestCase {
+  @Rule
+  public final DistributedRestoreSystemProperties restoreSystemProperties =
+      new DistributedRestoreSystemProperties();
+  private int locatorPort;
+
+  @Before
+  public void setup() throws IOException {
+    // Start a new locator with authorization
+    locatorPort = Host.getHost(0).getVM(0).invoke(() -> {
+      System.setProperty("geode.feature-protobuf-protocol", "true");
+      Properties props = new Properties();
+      props.setProperty(ConfigurationProperties.SECURITY_MANAGER,
+          SimpleTestSecurityManager.class.getName());
+      Locator locator = Locator.startLocatorAndDS(0, null, props);
+      return locator.getPort();
+    });
+
+    startCacheWithCacheServer(locatorPort);
+  }
+
+  private Socket createSocket() throws IOException {
+    Host host = Host.getHost(0);
+    Socket socket = new Socket(host.getHostName(), locatorPort);
+    MessageUtil.sendHandshake(socket);
+    MessageUtil.verifyHandshakeSuccess(socket);
+    return socket;
+  }
+
+  /**
+   * Test that if the locator has a security manager, an unauthorized client is not allowed to do
+   * anything.
+   */
+  @Test
+  public void authorizedClientCanGetServersIfSecurityIsEnabled() throws Throwable {
+    ClientProtocol.Request.Builder protobufRequestBuilder =
+        ProtobufUtilities.createProtobufRequestBuilder();
+
+    ClientProtocol.Message authorization =
+        ProtobufUtilities.createProtobufMessage(ProtobufUtilities.createProtobufRequestBuilder()
+            .setAuthenticationRequest(ConnectionAPI.AuthenticationRequest.newBuilder()
+                .putCredentials("security-username", "data").putCredentials("security-password",
+                    "data"))
+            .build());
+    ClientProtocol.Message getAvailableServersRequestMessage = ProtobufUtilities
+        .createProtobufMessage(protobufRequestBuilder.setGetAvailableServersRequest(
+            ProtobufRequestUtilities.createGetAvailableServersRequest()).build());
+
+    ProtobufProtocolSerializer protobufProtocolSerializer = new ProtobufProtocolSerializer();
+
+    try (Socket socket = createSocket()) {
+      protobufProtocolSerializer.serialize(authorization, socket.getOutputStream());
+
+      ClientProtocol.Message authorizationResponse =
+          protobufProtocolSerializer.deserialize(socket.getInputStream());
+      assertEquals(true,
+          authorizationResponse.getResponse().getAuthenticationResponse().getAuthenticated());
+      protobufProtocolSerializer.serialize(getAvailableServersRequestMessage,
+          socket.getOutputStream());
+
+      ClientProtocol.Message getAvailableServersResponseMessage =
+          protobufProtocolSerializer.deserialize(socket.getInputStream());
+      assertNotNull("Got response: " + getAvailableServersResponseMessage,
+          getAvailableServersRequestMessage.getResponse().getErrorResponse());
```

Comment:
Do we expect an error response here?  I'd have thought the test should be sending a getAvailableServersResponse after we authenticated.

Reply:
You are right. Looks like we are still getting an error response!